src/app/dataset-view/dataset.service.ts

```diff
@@ -226,9 +176,9 @@ export class AppDatasetService {
     ): void {
         this.searchApi
             .getDatasetOverview({ id, page })
-            .subscribe((data: DatasetOverviewQuery | undefined) => {
+            .subscribe((data: {datasets: Datasets} | undefined) => {
                 let datasets: SearchDatasetByID;
-                if (data) {
+                if (!_.isNil(data)) {
```

Comment:
I do not really understand what is going on here.
I see we are mapping GraphQL-generated type `Dataset` to hand-written type `SearchDatasetByID`. These data types have partial overlap of the fields, but they are not equal....

We should either use the `Dataset` type directly, or, if we still prever extra layer of wrapper, then it should be correctly modeled and mapped...